# report-caught-error

Require caught errors to be reported.

üíº This rule is enabled in the ‚úÖ `recommended` config.

üí° This rule is manually fixable by [editor suggestions](https://eslint.org/docs/latest/use/core-concepts#rule-suggestions).

‚öôÔ∏è This rule is configurable.

## Rule Details

Examples of **incorrect** code for this rule with the default options:

```js
try {
  // doing something try-catch worthy..
} catch (error) {
  let doingSomethingElse = true;
}
```

Examples of **correct** code for this rule with the default options:

```js
try {
  // doing something try-catch worthy.
} catch (error) {
  console.error(error);
}
```

## Options

This rule accepts the following options:

```ts
type Options = [
  {
    /**
     * The function name that is used to report errors.
     */
    reportFunctionName: string;

    /**
     * Whether or not to require the error identifier to be passed as first argument to the report function.
     */
    shouldPassErrorToFunction: boolean;
  },
];

const defaultOptions: Options = [
  {
    reportFunctionName: 'console.error',
    shouldPassErrorToFunction: true,
  },
];
```
