# require-useeffect-dependency-array

Require `useEffect` to have a dependency array.

ðŸ’¼ This rule is enabled in the âœ… `recommended` config.

ðŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

## Rule Details

Examples of **incorrect** code for this rule:

```js
useEffect(() => {});
```

Examples of **correct** code for this rule:

```js
useEffect(() => {}, []);
```
